src/algorithms/time_evolution/simpleupdate.jl

```diff
@@ -0,0 +1,207 @@
+"""
+Algorithm struct for simple update (SU) of infinite PEPS with bond weights/
+Each SU run is converged when the singular value difference becomes smaller than `tol`.
+"""
+struct SimpleUpdate
+    dt::Float64
+    tol::Float64
+    maxiter::Int
+    trscheme::TensorKit.TruncationScheme
+end
+
+function truncation_scheme(alg::SimpleUpdate, v::ElementarySpace)
+    if alg.trscheme isa FixedSpaceTruncation
+        return truncspace(v)
+    else
+        return alg.trscheme
+    end
+end
+
+"""
+Simple update of bond `peps.weights.x[r,c]`
+```
+                y[r,c]              y[r,c+1]
+                ↓                   ↓
+    x[r,c-1] ←- T[r,c] ←- x[r,c] ←- T[r,c+1] ← x[r,c+1]
+                ↓                   ↓
+                y[r+1,c]            y[r+1,c+1]
+```
+"""
+function _su_bondx!(
+    row::Int,
+    col::Int,
+    gate::AbstractTensorMap{S,2,2},
+    peps::InfiniteWeightPEPS,
+    alg::SimpleUpdate,
+) where {S}
+    Nr, Nc = size(peps)
+    @assert 1 <= row <= Nr && 1 <= col <= Nc
+    row2, col2 = row, _next(col, Nc)
+    T1, T2 = peps.vertices[row, col], peps.vertices[row2, col2]
+    # absorb environment weights
+    for ax in (2, 4, 5)
+        T1 = absorb_wt(T1, row, col, ax, peps.weights)
+    end
+    for ax in (2, 3, 4)
+        T2 = absorb_wt(T2, row2, col2, ax, peps.weights)
+    end
+    # absorb bond weight
+    T1 = absorb_wt(T1, row, col, 3, peps.weights; sqrtwt=true)
+    T2 = absorb_wt(T2, row2, col2, 5, peps.weights; sqrtwt=true)
+    #= QR and LQ decomposition
+
+        2   1               1             2
+        ↓ ↗                 ↓            ↗
+    5 ← T ← 3   ====>   3 ← X ← 4 ← 1 ← aR ← 3
+        ↓                   ↓
+        4                   2
+
+        2   1                 2         2
+        ↓ ↗                 ↗           ↓
+    5 ← T ← 3   ====>  1 ← bL ← 3 ← 1 ← Y ← 3
+        ↓                               ↓
+        4                               4
+    =#
+    X, aR = leftorth(T1, ((2, 4, 5), (1, 3)); alg=QRpos())
+    bL, Y = rightorth(T2, ((5, 1), (2, 3, 4)); alg=LQpos())
+    #= apply gate
+
+            -2          -3
+            ↑           ↑
+            |----gate---|
+            ↑           ↑
+            1           2
+            ↑           ↑
+        -1← aR -← 3 -← bL ← -4
+    =#
+    @tensor tmp[:] := gate[-2, -3, 1, 2] * aR[-1, 1, 3] * bL[3, 2, -4]
+    # SVD
+    aR, s, bL, ϵ = tsvd(tmp, ((1, 2), (3, 4)); trunc=truncation_scheme(alg, space(T1, 3)))
```

Comment:
Here, you can save a permutation/allocation by writing the output indices explicitly:
```suggestion
    @tensor tmp[-1 -2; -4 -3] := gate[-2, -3, 1, 2] * aR[-1, 1, 3] * bL[3, 2, -4]
    # SVD
    aR, s, bL, ϵ = tsvd!(tmp; trunc=truncation_scheme(alg, space(T1, 3)))
```